### PR TITLE
DPR2-239: Delete old code, rename and cleanup.

### PR DESCRIPTION
**Pull request by tom-ogle-moj**
- Delete old code.
- Delete unused/orphaned code.
- Renames
    - S3BatchProcessor to BatchProcessor
    - CuratedZoneLoadS3 to CuratedZoneLoad
    - CuratedZoneCDCS3 to CuratedZoneCDC
    - StructuredZoneLoadS3 to StructuredZoneLoad
    - StructuredZoneCDCS3 to StructuredZoneCDC
    - Remove S3 suffix from various methods in ViolationsService
- Remove BaseSparkTest and SparkTestHelpers classes that were duplicated in integration test from test. 
- Deleted some unused files in resources directories.
- Some additional tidy up and renaming.
    - E.g. Replace DMS_3_4_7.ParsedDataFields with CommonDataFields

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/job/batchprocessing/BatchProcessor.java [hunk 1]
```diff
@@ -31,95 +30,49 @@ public class BatchProcessor {
...
     private final StructuredZoneLoad structuredZoneLoad;
-    private final StructuredZoneCDC structuredZoneCDC;
     private final CuratedZoneLoad curatedZoneLoad;
-    private final CuratedZoneCDC curatedZoneCDC;
-    private final SourceReferenceService sourceReferenceService;
-    private final ViolationService violationService;
+    private final ValidationService validationService;
 
     @Inject
     public BatchProcessor(
```
Comment (tom-ogle-moj) [inline src/main/java/uk/gov/justice/digital/job/batchprocessing/BatchProcessor.java:38]: This is actually a delete of the old BatchProcessor and a rename of S3BatchProcessor to BatchProcessor so the diff is a bit misleading.

File: src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java [hunk 2]
```diff
@@ -2,11 +2,31 @@
...
+import static org.mockito.Mockito.when;
+import static uk.gov.justice.digital.common.CommonDataFields.ERROR_RAW;
+import static uk.gov.justice.digital.config.JobArguments.DATA_STORAGE_RETRY_JITTER_FACTOR_DEFAULT;
+import static uk.gov.justice.digital.config.JobArguments.DATA_STORAGE_RETRY_MAX_ATTEMPTS_DEFAULT;
+import static uk.gov.justice.digital.config.JobArguments.DATA_STORAGE_RETRY_MAX_WAIT_MILLIS_DEFAULT;
+import static uk.gov.justice.digital.config.JobArguments.DATA_STORAGE_RETRY_MIN_WAIT_MILLIS_DEFAULT;
+import static uk.gov.justice.digital.test.MinimalTestData.DATA_COLUMN;
+import static uk.gov.justice.digital.test.MinimalTestData.PRIMARY_KEY_COLUMN;
+
 public class BaseSparkTest {
```
Comment (tom-ogle-moj) [inline src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java:30]: Collapsed the duplicated class in to one

File: src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java [hunk 3]
```diff
@@ -42,4 +62,87 @@ public static void stopSession() {
 		spark.stop();
 	}
 
+	protected static void assertDeltaTableContainsPK(String tablePath, int primaryKey) {
```
Comment (tom-ogle-moj) [inline src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java:65]: These were just moved from the duplicated version of this class (which is now deleted)

File: src/test/java/uk/gov/justice/digital/test/SparkTestHelpers.java [hunk 4]
```diff
@@ -1,105 +1,66 @@
...
 import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static org.hamcrest.Matchers.contains;
 
 public class SparkTestHelpers {
```
Comment (tom-ogle-moj) [inline src/test/java/uk/gov/justice/digital/test/SparkTestHelpers.java:25]: This is another duplicated class (with some slight differences) where I've moved stuff from the duplicate in to here and deleted the duplicate.

**Review by dominic-messenger-moj — APPROVED** [review]

**Review by koladeadewuyi-moj — APPROVED** [review]

**Pull request merged**